idflakies-maven-plugin/src/main/java/edu/illinois/cs/dt/tools/plugin/IncDetectorMojo.java

```diff
@@ -0,0 +1,651 @@
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@Mojo(name = "incdetect", defaultPhase = LifecyclePhase.TEST, requiresDependencyResolution = ResolutionScope.TEST)
+public class IncDetectorMojo extends DetectorMojo {
+
+    protected static String CLASSES = "classes";
+    protected static String COMMA = ",";
+    protected static String JAR_CHECKSUMS = "jar-checksums";
+    protected static String SF_CLASSPATH = "sf-classpath";
+    protected static String TEST_CLASSES = "test-classes";
+    private static final String TARGET = "target";
+
+    /**
+     * The directory in which to store STARTS artifacts that are needed between runs.
+     */
+    protected String artifactsDir;
+
+    protected ClassLoader loader;
+
+    protected List<Pair> jarCheckSums = null;
+
+    protected Set<String> nonAffectedTests;
+
+    private Classpath sureFireClassPath;
+
+    protected boolean selectMore;
+
+    protected boolean detectOrNot;
+
+    protected String ekstaziSelectedTestsFile;
+
+    protected String ekstaziDependenciesFile;
+
+    protected String startsSelectedTestsFile;
+
+    protected String startsDependenciesFile;
+
+    protected boolean isEkstazi;
+
+    private Set<String> affectedTestClasses;
+
+    private static Set<String> immutableList;
+
+    @Override
+    public void execute() {
+        superExecute();
+
+        final ErrorLogger logger = new ErrorLogger();
+        this.outputPath = PathManager.detectionResults();
+        this.coordinates = mavenProject.getGroupId() + ":" + mavenProject.getArtifactId() + ":" + mavenProject.getVersion();
+
+        logger.runAndLogError(() -> defineSettings(logger, mavenProject));
+        if(this.runner == null) {
+            return;
+        }
+        long startTime = System.currentTimeMillis();
+        try {
+            affectedTestClasses = computeAffectedTests(mavenProject);
+            if (!detectOrNot) {
+                System.out.println("Not detect flaky tests at the first run");
+                return;
+            }
+        } catch (IOException | MojoExecutionException | ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+        timing(startTime);
+
+        startTime = System.currentTimeMillis();
+        logger.runAndLogError(() -> detectorExecute(logger, mavenProject, moduleRounds(coordinates)));
+        timing(startTime);
+    }
+
+    // from SelectMojo
+    private Set<String> computeAffectedTests(final MavenProject project) throws IOException, MojoExecutionException, ClassNotFoundException {
+        if (isEkstazi) {
```

Comment:
Is the reason to distinguish Ekstazi vs STARTS just because the format of dependencies differs? Can we get them to somehow simplify or consolidate some logic?